File: frontend/browser/lib/riceKeys.ts
// TODO:
// [ ] key factory: all = ["rice"] as const; project(id); role(r); model(role) — https://tanstack.com/query/latest/docs/framework/react/guides/query-keys
// [ ] invalidateProject(queryClient, id); invalidateAll(queryClient)
//
/** Namespaced Rice runtime keys — CHIEF hydrates values; SMITH serves db/ai routes. */

export const ricePrefixes = {
	db: "db.",
	ai: "ai.",
	ui: "ui.",
	i18n: "i18n.",
	feature: "feature.",
} as const;

export function riceDbKey(path: string): string {
	return `${ricePrefixes.db}${path}`;
}

export function riceAiKey(path: string): string {
	return `${ricePrefixes.ai}${path}`;
}

export function riceUiKey(path: string): string {
	return `${ricePrefixes.ui}${path}`;
}

export function riceI18nKey(path: string): string {
	return `${ricePrefixes.i18n}${path}`;
}

export function riceFeatureKey(path: string): string {
	return `${ricePrefixes.feature}${path}`;
}

export function isRemoteRiceKey(key: string): boolean {
	return (
		key.startsWith(ricePrefixes.db) || key.startsWith(ricePrefixes.ai)
	);
}

// TODO:
// [ ] Optional allowlist from env (e.g. NEXT_PUBLIC_RICE_KEY_ALLOWLIST) for production hardening.
// [ ] Align key vocabulary with proto messages in frontend/gen/ when MASON adds rice config protos.

File: frontend/screen/lib/rice_keys.dart
// TODO:
// [ ] const provider name strings; projectKey(id) cache keys — https://riverpod.dev/docs/concepts2/family
//
/// Namespaced Rice keys — same contract as [frontend/browser/lib/riceKeys.ts].

abstract final class RicePrefixes {
  static const db = 'db.';
  static const ai = 'ai.';
  static const ui = 'ui.';
  static const i18n = 'i18n.';
  static const feature = 'feature.';
}

String riceDbKey(String path) => '${RicePrefixes.db}$path';
String riceAiKey(String path) => '${RicePrefixes.ai}$path';
String riceUiKey(String path) => '${RicePrefixes.ui}$path';
String riceI18nKey(String path) => '${RicePrefixes.i18n}$path';
String riceFeatureKey(String path) => '${RicePrefixes.feature}$path';

bool isRemoteRiceKey(String key) =>
    key.startsWith(RicePrefixes.db) || key.startsWith(RicePrefixes.ai);

// TODO:
// [ ] Optional allowlist via dart-define for production key policies.
// [ ] Map keys to generated protos under frontend/gen/ when available.
